packages/core/README.md

```diff
@@ -69,3 +69,60 @@ const governor = await governor.deployed();
 
 Note: this use case is not particularly common, but it is sometimes useful to have access to multiple abi versions
 side-by-side.
+
+## Typescript!
+
+In addition to the above import styles, you can import typescript types for truffle, ethers, and web3. Because of existing
+limitations in typechain the import style for each of these is slightly different.
+
+Note: this is a work in progress and the typescript API will likely change and improve in the future.
+
+### Ethers
+
+The best support is for Ethers contract types. To construct an Ethers contract, simply import from the ethers factories:
+
+```ts
+import { Voting__factory } from "@uma/core/contract-types/ethers"
+
+// Alternative import style to avoid loading anything unnecessary
+// import { Voting__factory } from "@uma/core/contract-types/ethers/factories/Voting__factory"
+
+const provider = new ethers.providers.JsonRpcProvider(RPC_HOST)
+const votingInstance = Voting__factory.connect(VOTING_ADDRESS, provider)
+```
+
+If you just want the raw type, you can import as follows:
+
+```ts
+import type { Voting } from "@uma/core/contract-types/ethers";
+
+// Alternative import styles
+// import type { Voting } from "@uma/core/contract-types/ethers/Voting";
+```
+
+### Truffle
+
+Truffle has well-defined contract types as well, but there are no built-in truffle factories.
+
+```ts
+import type { VotingInstance, VotingContract } from "@uma/core/contract-types/truffle";
+
+// Alternative import style
+// import type { VotingInstance, VotingContract } from "@uma/core/contract-types/truffle/Voting";
+
+import { getTruffleContract } from "@uma/core";
+
+const Voting = getTruffleContract("Voting", web3) as VotingContract;
+const voting = Voting.deployed(); // Should be a VotingInstance.
+```
+
+### Web3
+
+Web3 types can be imported similarly to truffle. However, the import syntax is quite limited. There is no way to import
+all UMA web3 types from the same import. Each contract is specified in a separate file.
+
+```ts
+import type { Voting } from "@uma/core/contract-types/web3/Voting";
+
+const voting = new web3.eth.Contract(VOTING_ABI, VOTING_ADDRESS) as Voting;
```

Comment:
this syntax is awesome. as next steps we should refactor the typescript bots (trader specifically) to take advantage of this.

Reply:
Yep! I tried this out in some of the other packages and it seemed to work great!